Make fails when using paralellism

Should be a simple fix, but I'll have to leave this here as a reminder for now.

DEBUG: + ./configure
DEBUG: Checking for gcc...
DEBUG: Checking for shared library support...
DEBUG: Building shared library libz.so.1.2.8.zlib-ng with gcc.
DEBUG: Checking for off64_t... Yes.
DEBUG: Checking for fseeko... Yes.
DEBUG: Checking for strerror... Yes.
DEBUG: Checking for unistd.h... Yes.
DEBUG: Checking for stdarg.h... Yes.
DEBUG: Checking for ANSI C compliant compiler...  Yes.
DEBUG: Checking for attribute(visibility) support... Yes.
DEBUG: Checking for __builtin_ctzl ... Yes.
DEBUG: Checking for SSE2 intrinsics ... Yes.
DEBUG: Checking for PCLMULQDQ intrinsics ... Yes.
DEBUG: ARCH: x86_64
DEBUG: Using arch directory: arch/x86
DEBUG: + make -j7
DEBUG: make -C arch/x86
DEBUG: gcc -O3  -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY -I. -c -o example.o test/example.c
DEBUG: gcc -O3  -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY   -c -o adler32.o adler32.c
DEBUG: gcc -O3  -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY   -c -o crc32.o crc32.c
DEBUG: gcc -O3  -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY   -c -o deflate.o deflate.c
DEBUG: gcc -O3  -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY   -c -o match.o match.c
DEBUG: gcc -O3  -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY   -c -o infback.o infback.c
DEBUG: make[1]: Entering directory `/builddir/build/BUILD/zlib-ng-develop/arch/x86'
DEBUG: gcc -O3  -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY -I. -I../../ -c -o x86.o x86.c
DEBUG: gcc -O3  -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY -I. -I../../ -c -o x86.lo x86.c
DEBUG: gcc -O3  -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY   -c -o inffast.o inffast.c
DEBUG: gcc -O3  -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY -msse2 -I. -I../../ -c -o fill_window_sse.o fill_window_sse.c
DEBUG: gcc -O3  -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY   -c -o inflate.o inflate.c
DEBUG: gcc -O3  -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY   -c -o inftrees.o inftrees.c
DEBUG: gcc -O3  -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY   -c -o trees.o trees.c
DEBUG: gcc -O3  -fPIC -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY -msse2 -DPIC -I. -I../../ -c -o fill_window_sse.lo fill_window_sse.c
DEBUG: gcc -O3  -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY   -c -o zutil.o zutil.c
DEBUG: make: *** No rule to make target`x86.o', needed by `libz.a'.  Stop.
DEBUG: make: *** Waiting for unfinished jobs....
DEBUG: gcc -O3  -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY -msse4 -I. -I../../ -c -o deflate_quick.o deflate_quick.c
DEBUG: gcc -O3  -fPIC -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY -msse4 -DPIC -I. -I../../ -c -o deflate_quick.lo deflate_quick.c
DEBUG: gcc -O3  -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY -mpclmul -msse4 -I. -I../../ -c -o crc_folding.o crc_folding.c
DEBUG: gcc -O3  -fPIC -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN -DHAVE_BUILTIN_CTZL -DMEDIUM_STRATEGY -DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DADLER32_UNROLL_LESS -DCRC32_UNROLL_LESS -DX86_CPUID -DX86_SSE2_FILL_WINDOW -DX86_SSE4_2_CRC_HASH -DX86_PCLMULQDQ_CRC -DX86_QUICK_STRATEGY -mpclmul -msse4 -DPIC -I. -I../../ -c -o crc_folding.lo crc_folding.c
DEBUG: make[1]: Leaving directory`/builddir/build/BUILD/zlib-ng-develop/arch/x86'
DEBUG: mv arch/x86/_.o .
DEBUG: mv arch/x86/_.lo .

--- a/.gitignore
+++ b/.gitignore
@@ -37,3 +37,9 @@ CMakeCache.txt
 CMakeFiles
 Testing
 *.cmake
+
+configure.log
+
+/arch/arm/Makefile
+/arch/generic/Makefile
+/arch/x86/Makefile

--- a/Makefile.in
+++ b/Makefile.in
@@ -65,24 +65,27 @@ OBJS = $(OBJC)
 
 PIC_OBJS = $(PIC_OBJC)
 
-all: $(ARCHDIR) static shared
+all: static shared
 
-static: subdirs example$(EXE) minigzip$(EXE)
+static: example$(EXE) minigzip$(EXE)
 
-shared: subdirs examplesh$(EXE) minigzipsh$(EXE)
+shared: examplesh$(EXE) minigzipsh$(EXE)
 
-all64: subdirs example64$(EXE) minigzip64$(EXE)
+all64: example64$(EXE) minigzip64$(EXE)
 
 check: test
 
-.PHONY: subdirs $(ARCHDIR)
+$(ARCHDIR)/%.o: $(ARCHDIR)/%.c
+	$(MAKE) -C $(ARCHDIR) $(notdir $@)
 
-subdirs: $(ARCHDIR)
+$(ARCHDIR)/%.lo: $(ARCHDIR)/%.c
+	$(MAKE) -C $(ARCHDIR) $(notdir $@)
 
-$(ARCHDIR):
-	$(MAKE) -C $@
-	-mv $@/*.o .
-	-mv $@/*.lo .
+%.o : $(ARCHDIR)/%.o
+	-cp $< $@
+
+%.lo : $(ARCHDIR)/%.lo
+	-cp $< $@
 
 test: all teststatic testshared
 

